encrypted-media/scripts/requestmediakeysystemaccess.js

```diff
@@ -0,0 +1,311 @@
+        videoCapabilities: [{contentType: config.videoType.replace( /^(.*?codecs=\")(.*)/, "$1 $2")}],
+    }, 'Space at start of codecs parameter');
+
+    expect_config(config.keysystem, [{
+        videoCapabilities: [{contentType: config.videoType.replace( /^(.*?codecs=\".*)\"/, "$1 \"")}],
+    }], {
+        videoCapabilities: [{contentType: config.videoType.replace( /^(.*?codecs=\".*)\"/, "$1 \"")}],
+    }, 'Space at end of codecs parameter');
+
+    // contentType is not case sensitive (except the codec names).
+    expect_config(config.keysystem, [{
+        videoCapabilities: [{contentType: 'V'+config.videoType.substr(1)}],
+    }], {
+        videoCapabilities: [{contentType: 'V'+config.videoType.substr(1)}],
+    }, 'Video/' );
+
+    expect_config(config.keysystem, [{
+        videoCapabilities: [{contentType: config.videoType.replace( /^(.*?)c(odecs.*)/, "$1C$2")}],
+    }], {
+        videoCapabilities: [{contentType: config.videoType.replace( /^(.*?)c(odecs.*)/, "$1C$2")}],
+    }, 'Codecs=');
+
+    var t = config.videoType.match( /(.*?)(;.*)/ );
+    expect_config(config.keysystem, [{
+        videoCapabilities: [{contentType: t[1].toUpperCase()+t[2]}],
+    }], {
+        videoCapabilities: [{contentType: t[1].toUpperCase()+t[2]}],
+    }, 'Upper case MIME type');
+
+    t = config.videoType.match( /(.*?)codecs(.*)/ );
+    expect_config(config.keysystem, [{
+        videoCapabilities: [{contentType: t[1]+'CODECS'+t[2]}],
+    }], {
+        videoCapabilities: [{contentType: t[1]+'CODECS'+t[2]}],
+    }, 'CODECS=');
+
+    // Unrecognized attributes are not allowed.
+    expect_error(config.keysystem, [{
+      videoCapabilities: [{contentType: 'video/webm; foo="bar"'}],
+    }], 'NotSupportedError', 'Unrecognized foo with webm');
+
+    expect_error(config.keysystem, [{
+      videoCapabilities: [{contentType: 'video/mp4; foo="bar"'}],
+    }], 'NotSupportedError', 'Unrecognized foo with mp4');
+
+    expect_error(config.keysystem, [{
+      videoCapabilities: [{contentType: config.videoType+'; foo="bar"'}],
+    }], 'NotSupportedError', 'Unrecognized foo with codecs');
+
+    // Invalid contentTypes.
+    expect_error(config.keysystem, [{
+        videoCapabilities: [{contentType: 'fake'}],
+    }], 'NotSupportedError', 'contentType fake');
+
+    expect_error(config.keysystem, [{
+        audioCapabilities: [{contentType: 'audio/fake'}],
+    }], 'NotSupportedError', 'contentType audio/fake');
+
+    expect_error(config.keysystem, [{
+        videoCapabilities: [{contentType: 'video/fake'}],
+    }], 'NotSupportedError', 'contentType video/fake');
+
+    // The actual codec names are case sensitive.
+    t = config.videoType.match( /(.*?codecs=\")(.*?\")(.*)/ );
+    if ( t[2] !== t[2].toUpperCase() ) {
+        expect_error(config.keysystem, [{
+            videoCapabilities: [{contentType: t[1]+t[2].toUpperCase()+t[3] }],
+        }], 'NotSupportedError', 'codecs Vp8');
+    }
+
+    if ( t[2] !== t[2].toLowerCase() ) {
```

Comment:
I agree that it is not critical and can be cleaned up later, but it would probably be easier and take less time to clean it up while we are working with existing PRs. If it doesn't take a lot of time, I believe we should clean it up now. At least the issues that we've already found.​​​